Helpers.gs

```diff
@@ -833,26 +833,29 @@ function sendSummary() {
  */
 function callWithBackoff(func, maxRetries) {
   var tries = 0;
-  var timeout = 100;
   var result;
-  while(tries < maxRetries){
+  while(true){
     tries++;
     try{
       result = func();
       return result;
     }
     catch(e){
-      if (tries < maxRetries){
-        Logger.log(`Error, retrying in ${timeout}ms... [${e}]`);
-        Utilities.sleep(timeout);
-        timeout*=2; // Exponentially increase timeout
-      }
-      else {
+      if ( e.message.includes("Rate Limit Exceeded") ) {
```

Comment:
Is there a reason "Rate Limit Exceeded" immediately throws rather than retrying?

Reply:
The quotas are defined per day. In my opinion, it doesn't make sense to try again right away.

Reply:
Here's a good reference I found online: https://www.oreilly.com/library/view/going-gas/9781491940457/ch04.html In the section "Code for exponential backoff" they have a list of backoff error messages. "Rate Limit Exceeded" and "User Reate Limit Exceeded" are for the quota dealing with speed quotas and not daily quotas.

Reply:
Yes, you're right. I adapted the code to take that into account. I also added the random delay from your reference.